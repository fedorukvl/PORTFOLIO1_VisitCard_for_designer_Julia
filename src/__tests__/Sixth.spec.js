import React from "react";
import { shallow } from "enzyme";
import { Sixth } from "../Components/index.js";

describe("Tests of <Sixth/> component", () => {
	let sixthComponent;
	beforeEach(() => {
		sixthComponent = shallow(<Sixth />);
	});
	afterEach(() => {
		sixthComponent = null;
	});
	describe("First render of <Sixth/> component", () => {
		it("Renders properly", () => {
			expect(sixthComponent).not.toBeNull();
			expect(sixthComponent.debug()).toMatchSnapshot();
		});
	});

	describe("<Sixth/> component renders elements properly", () => {
		it("Renders elements propeply", () => {
			expect(sixthComponent.find(".finalDiv").children()).toHaveLength(2);
		});
	});

	describe("<Sixth/> component has <div .thankDiv> with <p> what includes `Если Вас заинтересовала моя кандидатура, то Вы можете связаться со мной несколькими способами:` text", () => {
		it("<div .thankDiv> has <p>", () => {
			expect(sixthComponent.find("p")).toHaveLength(1);
		});

		it("<div .thankDiv> has <p> with text", () => {
			expect(sixthComponent.find("p").text()).toEqual(
				"Если Вас заинтересовала моя кандидатура, то Вы можете связаться со мной несколькими способами:"
			);
		});
	});

	describe("<Sixth/> component has <div .linkDiv> with <a/> inside", () => {
		it("<div .linkDiv> has <a/>", () => {
			expect(sixthComponent.find(".linkDiv").find("a")).toHaveLength(2);
		});
	});
});
