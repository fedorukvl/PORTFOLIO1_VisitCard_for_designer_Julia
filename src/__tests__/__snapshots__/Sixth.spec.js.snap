// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <Sixth/> component First render of <Sixth/> component Renders properly 1`] = `
"<div className=\\"finalDiv\\">
  <div className=\\"leftImgDiv\\" />
  <div className=\\"thankDiv\\">
    <h1>
      Спасибо, что посмотрели мое портфолио ✌ 
    </h1>
    <p>
      Если Вас заинтересовала моя кандидатура, то Вы можете связаться со мной несколькими способами:
    </p>
    <div className=\\"linkDiv\\">
      <b>
        inst:
         
        <a href=\\"https://www.instagram.com/jjul_ash/\\">
          @jjul_ash
        </a>
      </b>
      <b>
        mail: 
        <a href=\\"jjul26@mail.ru\\">
          jjul26@mail.ru
        </a>
      </b>
    </div>
  </div>
</div>"
`;
